### PR TITLE
Updates for Julia 0.6

### PR DESCRIPTION
**Pull request by ararslan**
~~So far this only deals with the deprecations. I haven't figured out the `UnionAll`-related failure on 0.6 yet.~~

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,4 +1,5 @@
 julia 0.4
+Compat 0.19.0
```
Comment (tkelman) [inline REQUIRE]: 0.17 for abstract type

File: src/univariate.jl [hunk 2]
```diff
@@ -8,7 +8,7 @@ end
 kernel_dist(::Type{Normal},w::Real) = Normal(0.0,w)
 kernel_dist(::Type{Uniform},w::Real) = (s = 1.7320508075688772*w; Uniform(-s,s))
 
-typealias LocationScale Union{Laplace,Logistic,SymTriangularDist}
+@compat const LocationScale = Union{Laplace,Logistic,SymTriangularDist}
```
Comment (tkelman) [inline src/univariate.jl]: compat not needed

File: src/bivariate.jl [hunk 3]
```diff
@@ -79,7 +85,7 @@ function conv(k::BivariateKDE, dist::Tuple{UnivariateDistribution,UnivariateDist
     BivariateKDE(k.x, k.y, dens)
 end
 
-typealias BivariateDistribution Union{MultivariateDistribution,Tuple{UnivariateDistribution,UnivariateDistribution}}
+@compat const BivariateDistribution = Union{MultivariateDistribution,Tuple{UnivariateDistribution,UnivariateDistribution}}
```
Comment (tkelman) [inline src/bivariate.jl]: compat not needed

File: REQUIRE (hunk 1 above)
Comment (ararslan) [inline REQUIRE]: I thought this was using the indexing changes in 0.19 but I guess I was thinking of a different repo.

File: src/univariate.jl (hunk 2 above)
Comment (ararslan) [inline src/univariate.jl]: Yeah. Whoops.

File: REQUIRE (hunk 1 above)
Comment (tkelman) [inline REQUIRE]: ForwardDiff was using IndexStyle, this wasn't afaict

Comment (ararslan) [inline REQUIRE]: I was thinking Interpolations, which I fixed in relation to this fix, since that's a dependency

**Pull request merged**